### DISCUSSION TITLE
Unhandled exception when removing fuel rod from beacon station reactor

### DISCUSSION THREAD

**Discussion by Arctodites** [FakeFishGames/Barotrauma#14670] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

After successfully activating a beacon station, I took the fuel rod back out of the station's reactor (thus deactivating the beacon), and the game threw an unhandled exception: "Result 'UnexpectedError' was not handled by 'UnlockAchievements'".
![BeaconStationError](https://github.com/user-attachments/assets/e85b2696-f7b2-408f-98de-bf349dbd86e2)

Playing the latest Windows unstable version of Barotrauma (v. 1.6.9.0) via Steam build 1726604483 on Windows 10 Home.

### Reproduction steps

1.) Repair a beacon station that has a reactor.
2.) Power the station's reactor with a normal fuel rod.
3.) Activate the beacon station's sonar beacon.
4.) Remove the fuel rod from the station's reactor, or simply deactivate the sonar.
5.) The dev console should pop up with an error message about an unhandled exception.

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Single player

### Version

v1.6.4.0 (unstable)

### -

Barotrauma Unstable v. 1.6.9.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
"Result 'UnexpectedError' was not handled by 'UnlockAchievements.'"
See attached image file (.jpg).
```

**Comment by Regalis11** [comment]
Thank you for the report!

The achievement error seems unrelated to the beacon one, and we've got an existing report about it: https://github.com/FakeFishGames/Barotrauma/discussions/14434.

The beacon error is actually intentional, it's an error message only shown in debug and unstable builds when the beacon station goes inactive mid-mission. The intention is to diagnose the long-standing mystery bug of beacon missions failing even though the player activated the beacon, which we're suspecting to be caused by _something_ deactivating the beacon mid-round. I've now clarified the error message in our private repo to make it's intended purpose a little clearer.
